src/test/java/com/getindata/connectors/http/internal/table/lookup/querycreators/GenericJsonAndUrlQueryCreatorFactoryTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * © Copyright IBM Corp. 2025
+ */
+package com.getindata.connectors.http.internal.table.lookup.querycreators;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.catalog.*;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.factories.DynamicTableFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.getindata.connectors.http.LookupQueryCreator;
+import com.getindata.connectors.http.internal.table.lookup.HttpLookupConnectorOptions;
+import com.getindata.connectors.http.internal.table.lookup.LookupQueryInfo;
+import com.getindata.connectors.http.internal.table.lookup.LookupRow;
+import com.getindata.connectors.http.internal.table.lookup.RowDataSingleValueLookupSchemaEntry;
+import static com.getindata.connectors.http.internal.table.lookup.HttpLookupTableSourceFactory.row;
+import static com.getindata.connectors.http.internal.table.lookup.querycreators.GenericJsonAndUrlQueryCreatorFactory.*;
+
+class GenericJsonAndUrlQueryCreatorFactoryTest
+{
+    private Configuration config = new Configuration();
+
+    private DynamicTableFactory.Context tableContext;
+
+    @BeforeEach
+    public void setUp() {
+        CustomJsonFormatFactory.requiredOptionsWereUsed = false;
+        this.tableContext = getTableContext(this.config, ResolvedSchema.of(Column.physical("key1",
+                DataTypes.STRING())));
+    }
+
+    @Test
+    public void lookupQueryInfoTestStr() {
+        assertThat(CustomJsonFormatFactory.requiredOptionsWereUsed)
+                .withFailMessage(
+                        "CustomJsonFormat was not cleared, "
+                                + "make sure `CustomJsonFormatFactory.requiredOptionsWereUsed"
+                                + "= false` "
+                                + "was called before this test execution.")
+                .isFalse();
+        // GIVEN
+        this.config.setString("lookup-request.format", CustomJsonFormatFactory.IDENTIFIER);
+        this.config.setString(
+                String.format("lookup-request.format.%s.%s", CustomJsonFormatFactory.IDENTIFIER,
+                        CustomJsonFormatFactory.REQUIRED_OPTION), "optionValue");
+        this.config.set(REQUEST_QUERY_PARAM_FIELDS, List.of("key1"));
+        // WHEN/THEN
```

Comment:
nit: WHEN/THEN can be replaced with `EXPECT` (as in e.g. Spock framework).

Reply:
there is precedence for WHEN/THEN in the code base but not for EXPECT. I am inclined to keep it ias is. I you want to go though and change them all to EXPECT - maybe do this separately. WDYT?